src/llmcompressor/transformers/finetune/session_mixin.py

```diff
@@ -79,15 +79,20 @@ def __init__(
 
         # parse training and metadata args
         training_args = kwargs.get("args")
-        self.metadata = (
-            self._extract_metadata(
+
+        self.metadata = None
+        if training_args is not None:
+            # trl_sft_trainer pathway. Both training_args and data_args
+            # have `max_seq_length` which causes collision error
+            training_args_dict = training_args.to_dict()
+            if "max_seq_length" in training_args_dict:
```

Comment:
This is the only arg that we have this case for?

Reply:
Yes

Reply:
can you add a quick comment

Reply:
Ok